### ISSUE TITLE
[1.10.0]Conv2d grad bias gets wrong value for bfloat16 case 

### ISSUE THREAD

**Issue by XiaobingSuper** [pytorch/pytorch#68048]
## 🐛 Bug

For PyTorch [1.10.0](https://github.com/pytorch/pytorch/tree/v1.10.0), conv2d grad bias gets wrong value for bfloat16 case on cpu device.

## To Reproduce

1.  example code
```
import torch

import torch.nn as nn
import copy
x = torch.randn(1, 1, 80, 80)

m_fp32 = nn.Conv2d(1, 2, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1))
m_bf16 = copy.deepcopy(m_fp32).bfloat16()

x_bf16 = x.bfloat16()

y_fp32 = m_fp32(x).sum()
y_fp32.backward()

y_bf16 = m_bf16(x_bf16).sum()
y_bf16.backward()

print(m_fp32.bias.grad)
print(m_bf16.bias.grad)
```
2. running the code getting the following result
```
tensor([1600., 1600.])
tensor([256., 256.], dtype=torch.bfloat16)
```

## Expected behavior

For the bfloat16 case, the expected output should be
```
tensor([1600., 1600.])
tensor([1600., 1600.], dtype=torch.bfloat16)
```

 - PyTorch Version (1.10.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (source):
 - Build command you used (if compiling from source):
 - Python version: 3.8.12
 - CUDA/cuDNN version: None
 - GPU models and configuration: None
 - Any other relevant information:

## Additional context

The reason is that data overflow for bfloat16 reduced sum:
https://github.com/pytorch/pytorch/blob/36449ea93134574c2a22b87baad3de0bf8d64d42/aten/src/ATen/native/ConvolutionMM2d.cpp#L314-L319

for the reduced sum, we need to use accumulation type **float**  for the bfloat16 case.

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser

**Comment by XiaobingSuper** [comment]
@gchanan, PyTorch master hasn't this issue which https://github.com/pytorch/pytorch/pull/64280 using **at::sum_out** to compute **grad_bias**, so add it in the 1.10.1 milestone fornext release, thanks!

**Comment by zou3519** [comment]
hi pri for silent correctness

**Comment by jbschlosser** [comment]
Confirmed this was fixed in master. As mentioned above, #64280 refactored the bias gradient computation to use `sum_out` instead of manual accumulation.

**Comment by mruberry** [comment]
Notes from triage meeting:

@ngimel: @jbschlosser we should be sure our tests will catch this
@malfet: we should create a plan to do 10.1 or not and designate fixes for it
@ngimel: If we do 10.1 we should take this

**Comment by seemethere** [comment]
> Confirmed this was fixed in master. As mentioned above, #64280 refactored the bias gradient computation to use `sum_out` instead of manual accumulation.

Closing this since it's been confirmed to have been fixed in master and has a cherry pick created for the fix for the 1.10.1 release
